### PR TITLE
Open E.cash to non-developers

### PR DESCRIPTION
**Pull request by stitesExpensify**
### Details
This PR:
1. Removes the page from the sign-in flow where we were taking github usernames
2. Removes login checks that required a github username
3. Remove all the checks for canAccessExpensify which were blocking users without a GH or shared chat report

### Fixed Issues
Fixes 
https://github.com/Expensify/Expensify/issues/149098
https://github.com/Expensify/Expensify/issues/155153

### Tests
1. Create a new account
2. Make sure you are not asked for your GH username
3. Make sure you get a validation email
4. Click the email and make sure it sends you to e.cash (not e.com)
5. Log out and make sure you can log in normally
6. Log out and make sure you can log into an existing account

### Tested On

- [x] Web
- [ ] Mobile Web
- [x] Desktop
- [x] iOS
- [ ] Android

### CODE REVIEW

**Comment by stitesExpensify** [comment]
cc: @tgolen since you did a lot of this originally

File: src/pages/signin/LoginForm/LoginFormNarrow.js [hunk 1]
```diff
@@ -105,31 +103,13 @@ class LoginFormNarrow extends React.Component {
...
-                        {' '}
-                        <Text
-                            style={[styles.link, styles.mx1]}
-                            onPress={() => openURLInNewTab(CONST.UPWORK_URL)}
-                        >
-                            earn cash
-                        </Text>
-                        {' '}
-                        today!
+                        {'Money talks. And now that chat and payments are in one place, it\'s also easy. '}
```
Comment (alex-mechler) [inline src/pages/signin/LoginForm/LoginFormNarrow.js:111]: Does this need to be in quotes? Or can we just encode the apos?

Comment (alex-mechler) [inline src/pages/signin/LoginForm/LoginFormNarrow.js:111]: Does this need to be in quotes? Or can we just encode the apos?

File: src/pages/signin/LoginForm/LoginFormWide.js [hunk 2]
```diff
@@ -96,31 +94,13 @@ class LoginFormWide extends React.Component {
...
-                            {' '}
-                            <Text
-                                style={[styles.link, styles.mx1]}
-                                onPress={() => openURLInNewTab(CONST.UPWORK_URL)}
-                            >
-                                earn cash
-                            </Text>
-                            {' '}
-                            today!
+                            {'Money talks. And now that chat and payments are in one place, it\'s also easy. '}
```
Comment (alex-mechler) [inline src/pages/signin/LoginForm/LoginFormWide.js:102]: Same

**Review by alex-mechler — COMMENTED** [review]
Also, should this be on hold at all or are we ready for e.cash to be public? The only discussion I saw about it is that is was ready to go into review, but it didnt mention if we were ready to make e.cash open to the public.

File: src/pages/signin/LoginForm/LoginFormNarrow.js (hunk 1 above)
Comment (stitesExpensify) [reply src/pages/signin/LoginForm/LoginFormNarrow.js:111]: We can encode the apos if we want and then do a space like the one i removed

**Comment by stitesExpensify** [comment]
> Also, should this be on hold at all or are we ready for e.cash to be public? The only discussion I saw about it is that is was ready to go into review, but it didnt mention if we were ready to make e.cash open to the public.

Nope we are full send! Went off hold today 😄

**Comment by stitesExpensify** [comment]
https://expensify.slack.com/archives/C011W8BJ9L6/p1615501461160900?thread_ts=1615390233.108000&cid=C011W8BJ9L6

File: src/pages/signin/LoginForm/LoginFormNarrow.js (hunk 1 above)
Comment (alex-mechler) [reply src/pages/signin/LoginForm/LoginFormNarrow.js:111]: Personally, I think it is cleaner to do it with an encoded apos and an explict space.

Comment (stitesExpensify) [reply src/pages/signin/LoginForm/LoginFormNarrow.js:111]: Sounds good I'll fix it!

Comment (stitesExpensify) [reply src/pages/signin/LoginForm/LoginFormNarrow.js]: Updated

**Review by alex-mechler — APPROVED** [review]
Would love to also have @tgolen review this since it's been a while since I've been in this codebase

File: src/libs/API.js [hunk 3]
```diff
@@ -540,17 +539,6 @@ function ResendValidateCode(parameters) {
     return Network.post(commandName, parameters);
 }
 
-/**
- * @param {Object} parameters
- * @param {String} parameters.githubUsername
- * @returns {Promise}
- */
-function SetGithubUsername(parameters) {
```
[the remark this answers is not in the document]
Comment (stitesExpensify) [reply src/libs/API.js:548]: https://github.com/Expensify/Auth/pull/5258
https://github.com/Expensify/Web-Expensify/pull/30246

**Comment by stitesExpensify** [comment]
Going to pull master and make sure this still works given the `react-navigation` update. Should be fine but I want to double check

**Comment by stitesExpensify** [comment]
Yep still works!

**Review by tgolen — APPROVED** [review]

**Pull request merged**